tests/functional/helper.js

```diff
@@ -320,12 +320,14 @@ module.exports = {
       await this.wait();
 
       // We don't really want to go through the authentication flow because we
-      // are mocking everything.
+      // are mocking everything. So this next chunk of code manually
+      // makes a call to the DesktopAuthenticationListener to mock
+      // a successful authentication in browser.
       const url = await this.getLastUrl();
       const urlObj = new URL(url);
 
       const options = {
-        hostname: '127.0.0.1',
+        hostname: urlObj.hostname,
```

Comment:
Why not hardcode this as `localhost`? Maybe I don't understand `lastUrl` but it is probably a guardian url and that doesn't necessarily have the same host as the `DesktopAuthenticationListener.`